Refactor y mejora de la gestión de contratos

YanmirG opened the pull request:
Se eliminaron dependencias innecesarias en `Class1.cs` y se eliminó la clase `borrar`. Se actualizó el `ProjectGuid` en `capa_aplicacion.csproj` y se reorganizaron las referencias para mejorar la legibilidad. En `Contratos.cs`, se añadió el método `ObtenerContratosPorTrabajador` con manejo de excepciones. Se introdujo la clase `ServicioContratos` en `ServicioContratos.cs`, que encapsula la lógica de negocio para crear, actualizar, finalizar y consultar contratos, con validaciones para mejorar la robustez del sistema.

YanmirG committed:
Refactor y mejora de la gestión de contratos

Se eliminaron dependencias innecesarias en `Class1.cs` y se eliminó la clase `borrar`.
Se actualizó el `ProjectGuid` en `capa_aplicacion.csproj` y se reorganizaron las referencias para mejorar la legibilidad.
En `Contratos.cs`, se añadió el método `ObtenerContratosPorTrabajador` con manejo de excepciones.
Se introdujo la clase `ServicioContratos` en `ServicioContratos.cs`, que encapsula la lógica de negocio para crear, actualizar, finalizar y consultar contratos, con validaciones para mejorar la robustez del sistema.

Jonathan03R commented on capa_aplicacion/Servicios/ServicioContratos.cs:
```diff
@@ -0,0 +1,79 @@
+
+        public ServicioContratos()
+        {
+            _repo = new Contratos();
```
como parametro deberia pedir la conexion de sql server

Jonathan03R commented on capa_persistencia/modulo_principal/Contratos.cs:
```diff
@@ -108,5 +109,53 @@ public void ActualizarContrato(int contratoId, string usuario, string motivo, Co
             }
         }
 
```
hacer el procedimiento almacenado

Jonathan03R commented on capa_persistencia/modulo_principal/Contratos.cs:
```diff
@@ -108,5 +109,53 @@ public void ActualizarContrato(int contratoId, string usuario, string motivo, Co
+            }
+            finally
+            {
+                _accesoSQL.CerrarConexion();
```
No manipular las conexiones en la capa de persistencia, se hara en la capa de apliacion

Jonathan03R commented on capa_aplicacion/Servicios/ServicioContratos.cs:
```diff
@@ -0,0 +1,79 @@
+        }
+
+        // ✅ CREAR CONTRATO
+        public int CrearContrato(ContratoDTO contrato)
```
Iniciar transacciones y terminar , si hay error cancelar la transaccion

Jonathan03R replied:
@SoyAngelW

YanmirG committed:
Refactorización del servicio de contratos y mejoras UI

Se han realizado cambios significativos en `ServicioContratos.cs` para gestionar las conexiones a la base de datos mediante la nueva clase `AccesoSQLServer`. Se han añadido métodos para obtener listas de trabajadores, áreas, cargos, estados de contrato y tipos de pensión.

Se ha eliminado la referencia al proyecto `capa_presentacion` en `capa_aplicacion.csproj` y se ha añadido `ReferenciasDTO.cs` en `capa_dominio.csproj` con nuevas clases DTO.

En `Conexion.cs`, se ha actualizado la configuración de conexión para Azure SQL y se ha mejorado la gestión de errores y transacciones.

El controlador `HomeController` y sus vistas han sido eliminados, creando un nuevo `ContratosController` con una interfaz rediseñada en `Index.cshtml` para la gestión de contratos.

Se ha añadido un nuevo proyecto de pruebas `capa_pruebas` y se han implementado pruebas en `PruebaServicioContratos.cs`. También se han realizado cambios en las vistas de error y en el diseño general de la aplicación para mejorar la experiencia del usuario.

Jonathan03R committed:
Merge branch 'develop' into Capa_PresentacionV2
